[MVP-119]ML 서버 FastAPI 이전

juwon8891 opened the pull request:
### Issue Number
[MVP-119]ML 서버 FastAPI 이전
<!-- Jira Issue Number 기록. 이 때 스마트 커밋을 활용할 수 있음 -->
<!-- ex. close: MVP-9 -->

### 작업 내역
<!-- 커밋들의 변경점 설명 -->
ML 서빙 서버를 Flask -> FastAPI로 변경하였습니다.

### 변경사항
<!-- 코드의 구조가 근본적으로 바뀔 경우 표시 -->

### 작업 유형
- [x] 신규 기능 추가
- [ ] 버그 수정
- [ ] 리펙토링
- [ ] 문서 업데이트

### PR 특이 사항
<!-- 기타 설명 -->
속도 쪼끔 더 빨라짐 ㅎㅎ..

[MVP-119]: https://soma-mvp.atlassian.net/browse/MVP-119?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

woosukji committed:
change run script to print logs to shell

woosukji committed:
ready for fastapi

juwon8891 committed:
✨ [MVP-119] Flask -> FastAPI 이전 완료

woosukji committed:
✨ [MVP-119] Flask -> FastAPI 이전 완료

woosukji committed:
Merge branch 'juwon' of https://github.com/somamvp/voyager-ml-server into juwon

x
^x
^X

woosukji committed:
📝 [MVP] PR 템플릿 작성

juwon8891 committed:
📝 [MVP] PR 템플릿 작성

woosukji committed:
📝 [MVP] PR 템플릿 작성

juwon8891 committed:
[MVP] 유저변경

juwon8891 committed:
✅  [MVP] 테스트 코드 작성

juwon8891 committed:
Merge branch 'main' into juwon

juwon8891 committed:
[MVP] add: add yolov7 submodule

woosukji commented on model.py:
```diff
@@ -5,8 +5,8 @@
-
-  DEVICE = "cuda" if torch.cuda.is_available() else "cpu"
+  DEVICE = "cpu"
+  #DEVICE = "cuda" if torch.cuda.is_available() else "cpu"
```
이 부분은 당시 gpu 메모리가 없어서 cpu로 돌리려고 임시로 넣어놓은 줄이라 돌아가는 것이 나을듯합니다

woosukji commented on server_fast.py:
```diff
@@ -0,0 +1,61 @@
+
+  return_json = json.dumps(stateMachine.guides)
+
+  print(f"total runtime: {time.time() - tick}", flush=True)
```
`print()` 함수에서 `flush=true` 옵션을 줘야 로그파일에 출력할 때 기록이 제대로 되는데, 이를 위해서 함수 끝마다 옵션을 달고 `print` 해야 하는 조건이 추가되네요.

`server_guide_only.py` 에서는 이전에 이 문제를 해결하려고 `utils.py -> printf`를 오버라이딩해서 flush를 기본 옵션으로 하는 `print` 함수를 재정의했었는데, 이 방식 써보면 어떨까요

juwon8891 replied:
print() -> logger 라이브러리 사용하는 것으로 변경 했습니다.

woosukji commented on server_fast.py:
```diff
@@ -0,0 +1,61 @@
+from StateMachine import StateMachine
+from model import get_model
+import os, io, time, json
+
```
(아래 discussion 참고)
```suggestion
import utils.printf as print

```

woosukji reviewed (approved):
로깅에서 flush 문제에 관한 고민은 공유됐고,
나머지는 임시 주석코드의 존재라서
일단은 머지하고,
추후 변겅에서 클리닝 해주시면 될듯 합니당

테스팅 코드 붙으니 좋네요~~

woosukji committed:
[MVP] add: add requirements for yolov7

- platform-dependent 라이브러리들 삭제하고 yaml 간소화: 다른 os에서도 작동하도록
- 추후 yolov7으로 마이그레이션 하면서 잘 작동하는지 확인 필요

juwon8891 committed:
🔊 [MVP]로그 사용법 변경

juwon8891 committed:
Merge branch 'juwon' of https://github.com/somamvp/voyager-ml-server into juwon

juwon8891 commented:
코드 리뷰 통과로 머지합니다
